packages/lesswrong/components/posts/PostsPage/ImageCropPreview.tsx

```diff
@@ -0,0 +1,469 @@
+const initialWidth = 360 * 3;
+const aspectRatio = initialHeight / initialWidth;
+
+export type Coordinates = {
+  x: number,
+  y: number,
+  width: number,
+  height: number,
+}
+
+export type BoxCoordinates = Coordinates & { flipped: boolean }
+
+type BoxSubContainers = Record<CoordinatePosition, BoxCoordinates | null>
+
+type PositionedOffsets<T extends CoordinatePosition> = {
+  [k in `${T}XPct` | `${T}YPct` | `${T}WidthPct` | `${T}HeightPct`]: number;
+}
+
+function getOffsetPercentages<T extends CoordinatePosition>(imgCoordinates: Coordinates, boxCoordinates: BoxCoordinates, prefix: T): PositionedOffsets<T> {
+  const {
+    x: imgX,
+    y: imgY,
+    width: imgWidth,
+    height: imgHeight,
+  } = imgCoordinates;
+
+  const {
+    x: boxX,
+    y: boxY,
+    width: boxWidth,
+    height: boxHeight,
+  } = boxCoordinates;
+
+  const relativeXOffset = boxX - imgX;
+  const relativeYOffset = boxY - imgY;
+
+  const xPct = (relativeXOffset / imgWidth);
+  const yPct = (relativeYOffset / imgHeight);
+
+  const widthPct = (boxWidth / imgWidth);
+  // This is the "correct" height percentage
+  // const heightPct = (boxHeight / imgHeight);
+  // This is the "all the way to the bottom of the image" percentage, which we prefer to use if it turns out it doen't mess up aspect ratios or whatever
+  const heightPct = ((boxHeight - relativeYOffset) / boxHeight);
+
+  return {
+    [`${prefix}XPct`]: xPct,
+    [`${prefix}YPct`]: yPct,
+    [`${prefix}WidthPct`]: widthPct,
+    [`${prefix}HeightPct`]: heightPct
+  } as PositionedOffsets<T>;
+}
+
+const styles = (theme: ThemeType) => ({
+  button: {
+    padding: '10px 20px',
+    cursor: 'pointer',
+    backgroundColor: theme.palette.panelBackground.reviewGold,
+    color: 'white',
+    borderRadius: '4px',
+    fontSize: '1rem',
+  },
+  overlay: {
+    position: 'fixed',
+    backgroundColor: 'rgba(0, 0, 0, 0.7)',
+    top: 0,
+    left: 0,
+    width: '100%',
+    height: '100%',
+    zIndex: 10000,
+  },
+  moveableBox: { 
+    display: 'flex', 
+    flexDirection: 'column',
+    justifyContent: 'space-around',
+    marginBottom: '40px', 
+    position: 'absolute',
+    background: 'transparent',
+    border: '2px solid white',
+    cursor: 'move',
+    zIndex: 20000,
+  },
+  closeButton: {
+    position: 'absolute',
+    top: 0,
+    right: 0,
+    cursor: 'pointer',
+    padding: '2px 5px',
+    userSelect: 'none', // Prevent text selection
+    color: 'black',
+    backgroundColor: 'rgba(255, 255, 255, 0.7)',
+    fontSize: '1rem',
+  },
+  resizer: {
+    width: '10px',
+    height: '10px',
+    position: 'absolute',
+    bottom: 0,
+    right: 0,
+    cursor: 'nwse-resize',
+    backgroundColor: '#fff',
+    border: '1px solid #000',
+  },
+  saveAllCoordinates: {
+    position: 'absolute',
+    display: 'block',
+    bottom: 0,
+    left: 0,
+    width: '100%',
+    cursor: 'pointer',
+    padding: '2px 5px',
+    userSelect: 'none', // Prevent text selection
+    color: 'black',
```

Comment:
Will need to clean up all of the styles using absolute color values here to use `theme.palette` values (including, I think, in places like `border: '1px solid #000'`)